### DISCUSSION TITLE
After training multiple concepts, the textual inversions blend together

### DISCUSSION THREAD

**Discussion by taleksan** [d8ahazard/sd_dreambooth_extension#639] [General]
Say I am training two people, instance tokens "person_1" and "person_2" both of class type "man" in Dreambooth/SD. I give them the same classification images dir. 
After the model is trained, each individual subject is generated well, i.e. if I say "person_1 at work" and get a face of person_1, and if I say "person_2 at a club" I get a face of person two. 
However, if I say "person_1 and person_2 talking at work", they both get the same face, which is somewhere in the middle of the two source faces. Any ideas how to prevent this?

Thanks!

**Comment by d8ahazard** [comment]
Try "with" or "beside".

**Reply by taleksan** [reply]
You mean in txt2img prompt say "person_1 with person_2 ..."? Tried that, no luck, faces still get blended together in random ways. I feel like this has to do with their class tokens being the same word i.e. "man" for both during training. Now I am trying to train a "man" and a "woman" class together and see if I get the blending.

**Reply by taleksan** [reply]
So, an update - after training two persons of different classes - man and woman - the image blending no longer happens with txt2img output when prompted "person_1 and person_2 ..." and faces end up looking properly distinct. 

Not the question - how to archive this with multiple subjects of the same class? Or perhaps I can name one class "man" and another "guy"?

**Comment by Jellybit** [comment]
I've found that this is an issue with Stable Diffusion in general, and has been since the beginning. It has always frustrated me, and I dream of the day when it can handle it better. Asking for two celebrities will sometimes separate them, but often it will be a gradient between that and them being 50/50 merged. It's often difficult to get SD to understand/respect the language when asking for multiple subjects, though sometimes you get lucky.
